src/Query/ExecutorFactory.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+namespace React\Dns\Query;
+
+use React\Cache\ArrayCache;
+use React\Cache\CacheInterface;
+use React\Dns\Config\Config;
+use React\Dns\Config\HostsFile;
+use React\EventLoop\Loop;
+use React\EventLoop\LoopInterface;
+
+class ExecutorFactory
```

Comment:
This pretty much creates what most end users need to use this component. IMHO using `Factory` as name tells enough, and since this object holds no state, we can make all methods static.

Reply:
Never mind this comment's part on naming, missed that it is in the Query directory. But we can still make all methods static IMHO.

Reply:
hm, yes it is possible.... @clue ???

Reply:
No strong opinion here, but not sure I quite like the idea of going more into factory land. I understand it's not optimal, but perhaps adding a new method to the existing factory class helps reduce the API surface?

Reply:
in respect to the upcoming DI/DX in react, a more granular distribution of classes give the best opportunity to combine it in more fancy ways and a test-case can be more precisely. so we should it keep in two classes. the static way can be interest but not really needed, i would it leave as it is ...

Reply:
Been thinking about this over the weekend, and part of the issue of creating another factory is that we'd start putting them everywhere. And you have to start creating aliases for them if you use more than one. Not hard opinions here either.

Reply:
as long as the reactPHP libs does not depend and implement an DI/DX container, a factory is a good alternate. later we can  refactor to DI and can exchange or run in parallel mode.